include/openssl/core_dispatch.h

```diff
@@ -173,8 +174,9 @@ OSSL_CORE_MAKE_FUNC(int, BIO_vprintf, (OSSL_CORE_BIO *bio, const char *format,
                                        va_list args))
 OSSL_CORE_MAKE_FUNC(int, BIO_vsnprintf,
                    (char *buf, size_t n, const char *fmt, va_list args))
-OSSL_CORE_MAKE_FUNC(int, BIO_ctrl, (OSSL_CORE_BIO *bio,
-                                    int cmd, long num, void *ptr))
+/* OSSL_FUNC_BIO_CTRL_BOGUS doesn't have a corresponding declaration */
+OSSL_CORE_MAKE_FUNC(long, BIO_ctrl, (OSSL_CORE_BIO *bio,
+                                     int cmd, long num, void *ptr))
```

Comment:
This is still a breaking change, e.g. consider the following:

```` C
#include <openssl/core.h>
#include <openssl/core_dispatch.h>

static int (*c_bio_ctrl)(OSSL_CORE_BIO *bio, int cmd, long num, void *ptr) = NULL;

int OSSL_provider_init(const OSSL_CORE_HANDLE *handle, const OSSL_DISPATCH *in,
                       const OSSL_DISPATCH **out, void **provctx)
{
    for (; in->function_id != 0; in++) {
        if (in->function_id == OSSL_FUNC_BIO_CTRL)
            c_bio_ctrl = OSSL_FUNC_BIO_ctrl(in);
    }

    /* Do stuff */

    return 1;
}
````
Compiles just fine before this change. Afterwards we get:

````
ctrl.c: In function ‘OSSL_provider_init’:
ctrl.c:20:24: warning: assignment to ‘int (*)(OSSL_CORE_BIO *, int,  long int,  void *)’ {aka ‘int (*)(struct ossl_core_bio_st *, int,  long int,  void *)’} from incompatible pointer type ‘long int (*)(OSSL_CORE_BIO *, int,  long int,  void *)’ {aka ‘long int (*)(struct ossl_core_bio_st *, int,  long int,  void *)’} [-Wincompatible-pointer-types]
   20 |             c_bio_ctrl = OSSL_FUNC_BIO_ctrl(in);
      |                        ^
````